src/libraries/Microsoft.PowerFx.Core/Syntax/Visitors/DTypeVisitor.cs

```diff
@@ -14,6 +14,7 @@
 
 namespace Microsoft.PowerFx.Core.Syntax.Visitors
 {
+    // Visitor to convert AST to DType. 
```

Comment:
>// Visitor to convert AST to DType.  [](http://example.com/codeflow?start=4&length=36)

This is not quite right. If the AST is a single primitive node (i.e., StrLitNode, DecimalNode, BoolLitNode, DecimalNode, ...) then it returns DType.Invalid, which is not the corresponding type of the AST. Can you be more specific in what this class does?